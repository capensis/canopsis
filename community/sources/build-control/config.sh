# - Variable UPDATE_ETC_ACTION
# - Possible values : keep replace
#
# If defined, the user won't be asked what to do on config file update

UPDATE_ETC_ACTION="ask"

# - Variable MONGODB_ADMIN_PASSWD
# If defined, the user won't be asked to type a password for the MongoDB admin user

MONGODB_ADMIN_PASSWD="admin"

# - Variable MONGODB_USER
# If defined, the user won't be asked to enter a username for MongoDB

MONGODB_USER="cpsmongo"

# - Variable MONGODB_USER_PASSWD
# If defined, the user won't be asked to type a password for the MongoDB canopsis user

MONGODB_USER_PASSWD="canopsis"

# - Variable INFLUXDB_ADMIN_PASSWD
# If defined, the user won't be asked to type a password for the InfluxDB admin user

INFLUXDB_ADMIN_PASSWD="admin"

# - Variable INFLUXDB_USER
# If defined, the user won't be asked to enter a username for InfluxDB

INFLUXDB_USER="cpsinflux"

# - Variable INFLUXDB_USER_PASSWD
# If defined, the user won't be asked to type a password for the InfluxDB canopsis user

INFLUXDB_USER_PASSWD="canopsis"

# - Variable RABBITMQ_ADMIN_PASSWD
# If defined, the user won't be asked to type a password for the RabbitMQ admin user

RABBITMQ_ADMIN_PASSWD="admin"

# - Variable RABBITMQ_USER
# If defined, the user won't be asked to enter a username for RabbitMQ

RABBITMQ_USER="cpsrabbit"

# - Variable RABBITMQ_USER_PASSWD
# If defined, the user won't be asked to type a password for the RabbitMQ canopsis user

RABBITMQ_USER_PASSWD="canopsis"

# - Variable SSL_CA_PATH
# If defined and path exists, will copy the CA to Canopsis instead of generate a new one
# SSL_CA_PATH=""

# - Variable SSL_CAKEY_PATH
# If defined and path exists, will copy the CA key to Canopsis instead of generate a new one
# SSL_CAKEY_PATH=""

# - Variable SSL_KEY_PASS
# Password for CA certificate key if we are generating one. Will prompt the user if empty
SSL_KEY_PASS="cpsnode"

# - Variable SSL_KEY_BITS
# Key size in bits
SSL_KEY_BITS="1024"

# - Variable SSL_CHECK_SECONDS
# Range used to check for certificate expiration from now
SSL_CHECK_SECONDS="86400"
