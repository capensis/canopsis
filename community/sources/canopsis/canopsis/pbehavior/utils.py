"""
Utils for pbehaviors.
"""

from dateutil.rrule import rrulestr
from dateutil import tz
from datetime import datetime


EXDATE_DATE_FORMAT = "%Y/%m/%d %H:%M:%S"


def check_valid_rrule(rrule):
    """
    Check for RRULE validity.

    Raises error when invalid with a more explicit message.
    :param str rrule: rrule as string
    :raises ValueError: rrule is invalid
    :returns: True if ok, raises exception if invalid
    :rtype: bool
    """
    if rrule == '' or rrule is None:
        return True

    try:
        rrulestr(rrule)
    except ValueError as ex:
        raise ValueError('Invalid RRULE: {}'.format(ex))

    return True
