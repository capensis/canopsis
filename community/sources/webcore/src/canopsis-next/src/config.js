import { removeTrailingSlashes } from '@/helpers/url';

export const {
  BASE_URL,
  VUE_APP_API_HOST,
  VUE_APP_LOCAL_STORAGE_ACCESS_TOKEN_KEY,
  VUE_APP_PAGINATION_LIMIT,
  VUE_APP_OPEN_STREET_LAYER_URL,
} = process.env;

export const APP_HOST = removeTrailingSlashes(`${window.location.origin}${BASE_URL}`);

export const API_HOST = VUE_APP_API_HOST || window.location.origin;

export const SOCKET_HOST = API_HOST.replace(/^http(s?)/, 'wss');

export const SOCKET_ROUTE = '/api/v4/ws';

export const SOCKET_URL = removeTrailingSlashes(`${SOCKET_HOST}${SOCKET_ROUTE}`);

export const OPEN_STREET_LAYER_URL = VUE_APP_OPEN_STREET_LAYER_URL;

export const ROUTER_MODE = 'history';

export const ROUTER_ACCESS_TOKEN_KEY = 'access_token';

export const LOCAL_STORAGE_ACCESS_TOKEN_KEY = VUE_APP_LOCAL_STORAGE_ACCESS_TOKEN_KEY || 'accessToken';

export const PAGINATION_LIMIT = parseInt(VUE_APP_PAGINATION_LIMIT, 10);

export const PAGINATION_PER_PAGE_VALUES = [5, 10, 20, 50, 100];

export const PAGINATION_TOTAL_VISIBLE = 7;

export const DEFAULT_MAX_MULTI_SORT_COLUMNS_COUNT = 3;

export const DEFAULT_WEATHER_LIMIT = 120;

export const DEFAULT_LOCALE = 'en';

export const DEFAULT_VIEW_STATS_INTERVAL = 120000;

export const LOCALE_PRIORITIES = {
  default: 1,
  global: 2,
  personal: 3,
};

export const POPUP_AUTO_CLOSE_DELAY = 3000;

export const ZOOM_OVERLAY_DELAY = 1500;

export const VUETIFY_ANIMATION_DELAY = 300;

export const POPUP_TICK_DELAY = 100;

export const SIDE_BAR_WIDTH = 250;

export const TOP_BAR_HEIGHT = 48;

export const EXPAND_DEFAULT_MAX_LETTERS = 50;

export const SCHEMA_EMBEDDED_KEY = '_embedded';

export const MEDIA_QUERIES_BREAKPOINTS = {
  m: 450,
  t: 900,
  l: 1400,
  xl: 'Infinity',
};

export const MAX_LOGO_SIZE_IN_KB = 16384;

export const DEFAULT_APP_TITLE = 'Canopsis';

export const EXPORT_VIEWS_AND_GROUPS_FILENAME_PREFIX = 'canopsis_groups_views-';

export const SOCKET_ROOMS = {
  healthcheck: 'healthcheck',
  healthcheckStatus: 'healthcheck-status',
  messageRates: 'message-rates',
  loggedUserCount: 'logged-user-count',
  broadcastMessages: 'broadcast-messages',
  execution: 'execution',
  declareticket: 'declareticket',
  alarms: 'alarms',
  alarmDetails: 'alarm-details',
};

export const API_ROUTES = {
  login: '/api/v4/login',
  logout: '/api/v4/logout',
  loggedUserCount: '/api/v4/logged-user-count',
  currentUser: '/api/v4/account/me',
  maintenance: '/api/v4/maintenance',
  alarms: {
    list: '/api/v4/alarms',
    bulkList: '/api/v4/bulk/alarms',
  },
  componentAlarms: '/api/v4/component-alarms',
  resolvedAlarms: '/api/v4/resolved-alarms',
  alarmDetails: '/api/v4/alarm-details',
  openAlarms: '/api/v4/open-alarms',
  alarmLinks: '/api/v4/alarm-links',
  entity: '/api/v4/entities',
  entityContextGraph: '/api/v4/entities/context-graph',
  bulkEntitiesEnable: '/api/v4/bulk/entities/enable',
  bulkEntitiesDisable: '/api/v4/bulk/entities/disable',
  entityBasics: '/api/v4/entitybasics',
  service: '/api/v4/entityservices',
  serviceDependencies: '/api/v4/entityservice-dependencies',
  serviceImpacts: '/api/v4/entityservice-impacts',
  serviceAlarms: '/api/v4/entityservice-alarms',
  entityInfosKeys: '/api/v4/entity-infos-dictionary/keys',
  weatherService: '/api/v4/weather-services',
  alarmListExport: '/api/v4/alarm-export',
  alarmTag: {
    list: '/api/v4/alarm-tags',
    bulkList: '/api/v4/bulk/alarm-tags',
  },
  contextExport: '/api/v4/entity-export',
  event: '/api/v4/event',
  userPreferences: '/api/v4/user-preferences',
  view: {
    list: '/api/v4/views',
    copy: '/api/v4/view-copy',
    export: '/api/v4/view-export',
    import: '/api/v4/view-import',
    positions: '/api/v4/view-positions',
    groups: '/api/v4/view-groups',
    tabs: '/api/v4/view-tabs',
    tabCopy: '/api/v4/view-tab-copy',
    tabPositions: '/api/v4/view-tab-positions',
  },
  widget: {
    list: '/api/v4/widgets',
    gridPositions: '/api/v4/widget-grid-positions',
    filters: '/api/v4/widget-filters',
    filterPositions: '/api/v4/widget-filter-positions',
  },
  widgetTemplate: '/api/v4/widget-templates',
  permissions: '/api/v4/permissions',
  users: '/api/v4/users',
  roles: {
    list: '/api/v4/roles',
    templates: '/api/v4/role-templates',
  },
  eventFilter: {
    list: '/api/v4/eventfilter',
    rules: '/api/v4/eventfilter/rules',
  },
  file: '/api/v4/file',
  fileAccess: '/api/v4/file-access',
  infos: {
    app: '/api/v4/app-info',
    userInterface: '/api/v4/internal/user_interface',
  },
  associativeTable: '/api/v4/associativetable',
  sessionsCount: '/api/v4/sessions-count',
  broadcastMessage: {
    list: '/api/v4/broadcast-message',
    activeList: '/api/v4/active-broadcast-message',
  },
  counter: '/api/v4/alarm-counters',
  playlist: '/api/v4/playlists',
  pbehavior: {
    timespan: '/api/v4/pbehavior-timespans',
    exceptions: '/api/v4/pbehavior-exceptions',
    exceptionImport: '/api/v4/pbehavior-exception-import',
    types: '/api/v4/pbehavior-types',
    nextTypesPriority: '/api/v4/pbehavior-types/next-priority',
    pbehaviors: '/api/v4/pbehaviors',
    bulkPbehaviors: '/api/v4/bulk/pbehaviors',
    pbehaviorComments: '/api/v4/pbehavior-comments',
    entities: '/api/v4/entities/pbehaviors',
    entitiesCalendar: '/api/v4/entities/pbehavior-calendar',
    reasons: '/api/v4/pbehavior-reasons',
    calendar: '/api/v4/pbehavior-calendar',
    bulkEntityPbehaviors: '/api/v4/bulk/entity-pbehaviors',
  },
  engineRunInfo: '/api/v4/engine-runinfo',
  cas: {
    login: '/api/v4/cas/login',
    loggedin: '/api/v4/cas/loggedin',
  },
  saml: {
    auth: '/api/v4/saml/auth',
  },
  scenario: {
    scenarios: '/api/v4/scenarios',
  },
  entityCategories: '/api/v4/entity-categories',
  stateSetting: '/api/v4/state-settings',
  dataStorage: '/api/v4/data-storage',
  notification: '/api/v4/notification',
  idleRules: '/api/v4/idle-rules',
  flappingRules: '/api/v4/flapping-rules',
  resolveRules: '/api/v4/resolve-rules',
  messageRateStats: '/api/v4/message-rate-stats',
  pattern: {
    list: '/api/v4/patterns',
    bulkList: '/api/v4/bulk/patterns',
    entitiesCount: '/api/v4/patterns-entities-count',
    alarmsCount: '/api/v4/patterns-alarms-count',
  },
  shareTokens: '/api/v4/share-tokens',
  techMetrics: '/api/v4/tech-metrics-export',
  templateVars: '/api/v4/template-vars',
  templateValidator: {
    declareTicketRules: '/api/v4/template-validator/declare-ticket-rules',
    scenarios: '/api/v4/template-validator/scenarios',
    eventFilterRules: '/api/v4/template-validator/event-filter-rules',
  },
  linkRule: '/api/v4/link-rules',
  linkCategories: '/api/v4/link-categories',
  themes: {
    list: '/api/v4/color-themes',
    bulkList: '/api/v4/bulk/color-themes',
  },
  healthcheck: {
    engines: '/api/v4/healthcheck',
    status: '/api/v4/healthcheck/status',
    enginesOrder: '/api/v4/healthcheck/engines-order',
    parameters: '/api/v4/healthcheck/parameters',
  },

  /**
   * Cat routes
   */
  filters: '/api/v4/cat/kpi-filters',
  ratingSettings: '/api/v4/cat/rating-settings',
  bulkRatingSettings: '/api/v4/cat/rating-settings/bulk',
  dynamicInfo: '/api/v4/cat/dynamic-infos',
  dynamicInfosDictionaryKeys: '/api/v4/cat/dynamic-infos-dictionary/keys',
  metaAlarmRule: '/api/v4/cat/metaalarmrules',
  remediation: {
    instructions: '/api/v4/cat/instructions',
    instructionStats: '/api/v4/cat/instruction-stats',
    jobs: '/api/v4/cat/jobs',
    configurations: '/api/v4/cat/job-configs',
    executions: '/api/v4/cat/executions',
    jobExecutions: '/api/v4/cat/job-executions',
    pausedExecutions: '/api/v4/cat/account/paused-executions',
    instructionComments: '/api/v4/cat/instruction-comments',
  },
  junit: {
    directory: '/api/v4/cat/junit/directory',
    testSuites: '/api/v4/cat/junit/test-suites',
    entityGantt: '/api/v4/cat/junit/test-suites-entity-gantt',
    history: '/api/v4/cat/junit/test-suites-history',
    widget: '/api/v4/cat/junit/test-suites-widget',
    file: '/api/v4/cat/junit/test-cases-file',
  },
  metrics: {
    settings: '/api/v4/cat/metrics-settings',
    alarm: '/api/v4/cat/metrics/alarm',
    exportAlarm: '/api/v4/cat/metrics-export/alarm',
    exportRating: '/api/v4/cat/metrics-export/rating',
    exportSli: '/api/v4/cat/metrics-export/sli',
    exportAggregate: '/api/v4/cat/metrics-export/aggregate',
    exportRemediation: '/api/v4/cat/metrics-export/remediation',
    exportMetric: '/api/v4/cat/metrics-export',
    sli: '/api/v4/cat/metrics/sli',
    rating: '/api/v4/cat/metrics/rating',
    remediation: '/api/v4/cat/metrics/remediation',
    aggregate: '/api/v4/cat/metrics/aggregate',
    perfDataMetrics: '/api/v4/cat/perf-data-metrics',
    entityAlarmMetrics: '/api/v4/cat/entity-metrics/alarm',
    entityAggregateMetrics: '/api/v4/cat/entity-metrics/aggregate',
    group: '/api/v4/cat/metrics/group',
    exportGroup: '/api/v4/cat/metrics-export/group',
  },
  maps: '/api/v4/cat/maps',
  bulkMaps: '/api/v4/cat/maps/bulk',
  mapState: '/api/v4/cat/map-state',
  manualMetaAlarm: '/api/v4/cat/manual-meta-alarms',
  metaAlarm: '/api/v4/cat/meta-alarms',
  snmpRule: '/api/v4/cat/snmprules',
  snmpMib: '/api/v4/cat/snmpmibs',
  declareTicket: {
    rules: '/api/v4/cat/declare-ticket-rules',
    bulkRules: '/api/v4/cat/bulk/declare-ticket-rules',
    alarmsAssigned: '/api/v4/cat/declare-ticket-assigned',
    testExecution: '/api/v4/cat/test-declare-ticket-executions',
    testExecutionWebhooks: '/api/v4/cat/test-declare-ticket-webhooks',
    declareTicketExecution: '/api/v4/cat/declare-ticket-executions',
    bulkDeclareTicket: '/api/v4/cat/bulk/declare-ticket-executions',
  },
  tags: '/api/v4/cat/tags',
};

export const COLORS = {
  primary: '#2fab63',
  secondary: '#2b3e4f',
  accent: '#82b1ff',
  error: '#ff5252',
  info: '#2196f3',
  success: '#4caf50',
  warning: '#fb8c00',
  background: '#ffffff',
  activeColor: '#000000',
  table: {
    background: '#FFFFFF',
    rowColor: '#FFFFFF',
    shiftRowColor: '#F5F5F5',
    hoverRowColor: '#F5F5F5',
  },
  state: {
    ok: '#00a65a',
    minor: '#fcdc00',
    major: '#ff9900',
    critical: '#f56954',
    pause: '#808080',
  },
  status: {
    closed: '#2fab63',
    stealthy: '#757575',
    flapping: '#2b3E4F',
    ongoing: '#f9A825',
    cancelled: '#757575',
    noEvents: '#ff5252',
    unknown: 'black',
  },
  entitiesEvents: {
    ack: '#9c27b0',
    ackRemove: '#9c27b0',
    declareTicket: '#2196f3',
    snooze: '#e91e63',
    pbhenter: '#18ffff',
    pbhleave: '#18ffff',
    comment: '#ba68c8',
    metaalarmattach: '#6d4c41',
  },
  testSuiteStatuses: {
    passed: '#2fab63',
    error: '#ffa800',
    failed: '#ff5252',
    skipped: '#5a6D80',
  },
  healthcheck: {
    error: '#FF5252',
    warning: '#FFA800',
    unknown: '#C4C4C4',
    edgeGray: '#979797',
    edgeBlack: '#000000',
  },
  impactState: [
    '#2FAB63',
    '#7CB342',
    '#CDDC39',
    '#DCE775',
    '#FFF59D',
    '#FFF176',
    '#FFEB3B',
    '#FDD835',
    '#FBC02D',
    '#FFB300',
    '#F9A825',
    '#FF8F00',
    '#F57C00',
    '#FF6F00',
    '#EF6C00',
    '#FF6A3C',
    '#FF5722',
    '#F4511E',
    '#F94C16',
    '#F44336',
    '#E64A19',
    '#D84315',
    '#E53935',
    '#D32F2F',
    '#C62828',
    '#BF360C',
    '#B71C1C',
    '#C51162',
    '#C2185B',
    '#AD1457',
    '#880E4F',
  ],
  kpi: {
    downtime: '#c4c4c4',
    maintenance: '#ffa800',
    uptime: '#5b6E7f',
  },
  metrics: {
    activeAlarms: '#5a6d7e',
    createdAlarms: '#fda701',
    nonDisplayedAlarms: '#fd693b',
    instructionAlarms: '#7bb242',
    pbehaviorAlarms: '#d64315',
    correlationAlarms: '#fdef75',
    ackAlarms: '#fd5252',
    cancelAckAlarms: '#9b27af',
    ackActiveAlarms: '#81b0fd',
    ticketActiveAlarms: '#2faa63',
    withoutTicketActiveAlarms: '#747474',
    ratioCorrelation: '#c31162',
    ratioInstructions: '#cbda39',
    ratioTickets: '#2195f1',
    ratioNonDisplayed: '#f9bf2d',
    averageAck: '#f5c6ab',
    averageResolve: '#1afd01',
    totalUserActivity: '#1fbbd1',
    manualInstructionExecutedAlarms: '#2FAB63',
    ratioRemediatedAlarms: '#FFA800',
    remediationStatisticAssignedRemediations: '#FFA800',
    remediationStatisticExecutedRemediations: '#5B6E7F',
    remediationStatisticRatioRemediations: '#5B6E7F',
    notAckedAlarms: '#dce775',
    notAckedInHourAlarms: '#1fbcd3',
    notAckedInFourHoursAlarms: '#afb42b',
    notAckedInDayAlarms: '#fff176',
  },
  mermaid: {
    primaryColor: '#bfe4ce',
    primaryBorderColor: '#2faa62',
    textColor: '#323232',
    lineColor: '#323232',
    noteBkgColor: '#75818c',
    noteTextColor: '#fdfdfd',
  },
  flowchart: {
    selection: '#5b6e7f',
    shapes: [
      '#fd897f',
      '#fd7faa',
      '#e87ffa',
      '#b287fd',
      '#8b9dfd',
      '#7fd6fd',
      '#83fdfd',
      '#a6fde9',
      '#b8f4c8',
      '#cafd8f',
      '#f2fd80',
      '#fdfd8c',
      '#fde37e',
      '#fdcf7f',
      '#fd9d7f',
      '#d5cac6',
      '#f3f3f3',
      '#cdd6da',
    ],
    border: [
      '#fd1743',
      '#f30056',
      '#d300f7',
      '#641ffd',
      '#2978fd',
      '#00affd',
      '#00b7d2',
      '#1de7b5',
      '#00e475',
      '#75fd03',
      '#c4fd00',
      '#fde800',
      '#fdc200',
      '#fd9000',
      '#fd3d00',
      '#8c6d62',
      '#9d9d9d',
      '#5f7c8a',
    ],
    text: [
      '#b61c1c',
      '#870e4e',
      '#49148b',
      '#311b91',
      '#0d46a0',
      '#01569a',
      '#005f63',
      '#004c3f',
      '#1b5d20',
      '#33681e',
      '#9d9c24',
      '#f37e17',
      '#fd6e00',
      '#e45000',
      '#be360c',
      '#4d342e',
      '#414141',
      '#37464e',
    ],
    background: [
      '#ffffff',
      '#eceff1',
      '#fafafa',
      '#efebe9',
      '#fbe9e7',
      '#f9fbe7',
      '#fff8e1',
      '#fffde7',
      '#fff3e0',
      '#f1f8e9',
      '#e0f2f1',
      '#e0f7fa',
      '#e8f5e9',
      '#f3e5f5',
      '#ede7f6',
      '#e8eaf6',
      '#e3f2fd',
      '#fce4ec',
      '#ffebee',
    ],
  },
};

export const CSS_COLORS_VARS = {
  primary: 'var(--v-primary-base)',
  secondary: 'var(--v-secondary-base)',
  error: 'var(--v-error-base)',

  state: {
    ok: 'var(--v-state-ok-base)',
    minor: 'var(--v-state-minor-base)',
    major: 'var(--v-state-major-base)',
    critical: 'var(--v-state-critical-base)',
    pause: 'var(--v-state-pause-base)',
  },

  status: {
    closed: 'var(--v-status-closed-base)',
    stealthy: 'var(--v-status-stealthy-base)',
    flapping: 'var(--v-status-flapping-base)',
    ongoing: 'var(--v-status-ongoing-base)',
    cancelled: 'var(--v-status-cancelled-base)',
    noEvents: 'var(--v-status-noEvents-base)',
  },
};

export const DEFAULT_THEME_COLORS = {
  primary: COLORS.primary,
  secondary: COLORS.secondary,
  accent: COLORS.accent,
  error: COLORS.error,
  info: COLORS.info,
  success: COLORS.success,
  warning: COLORS.warning,
  background: COLORS.background,
  active_color: COLORS.activeColor,

  state: COLORS.state,
  status: COLORS.status,
  table: {
    background: COLORS.table.background,
    row_color: COLORS.table.rowColor,
    hover_row_color: COLORS.table.hoverRowColor,
  },
};

export const FILE_BASE_URL = `${API_HOST}${API_ROUTES.file}`;

export const DOCUMENTATION_BASE_URL = 'https://doc.canopsis.net/';

export const EXPORT_FETCHING_INTERVAL = 2000;

export const MAX_PBEHAVIOR_DATES_DIFF_YEARS = 5;

export const HEALTHCHECK_HISTORY_FILENAME_PREFIX = 'healthcheck_history-';

export const HEALTHCHECK_LAST_HOUR_FILENAME_PREFIX = 'healthcheck_last_hour-';

export const KPI_ALARM_METRICS_FILENAME_PREFIX = 'kpi_alarm_metrics';

export const KPI_RATING_METRICS_FILENAME_PREFIX = 'kpi_rating_metrics';

export const KPI_SLI_METRICS_FILENAME_PREFIX = 'kpi_sli_metrics';

export const REMEDIATION_STATISTICS_FILENAME_PREFIX = 'remediation_statistics';

export const APP_INFO_FETCHING_INTERVAL = 10000;

export const MIN_CLICK_OUTSIDE_DELAY_AFTER_REGISTERED = 100;
