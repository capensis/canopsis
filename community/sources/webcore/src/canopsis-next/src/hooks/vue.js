import { getCurrentInstance } from 'vue';

/**
 * Hook for get current component instance
 *
 * @return {Vue}
 */
export const useComponentInstance = () => {
  const instance = getCurrentInstance();

  return instance?.proxy || instance;
};

/**
 * Hook for get current component options
 *
 * @return {Object}
 */
export const useComponentOptions = () => {
  const instance = useComponentInstance();

  return instance.$options;
};

/**
 * Hook for get current component model
 *
 * @return {Object}
 */
export const useComponentModel = () => {
  const { model } = useComponentOptions();

  return {
    event: model?.event ?? 'input',
    prop: model?.event ?? 'value',
  };
};
