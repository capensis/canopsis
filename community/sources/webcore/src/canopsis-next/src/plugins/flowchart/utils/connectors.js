import { range } from 'lodash';

import uid from '@/helpers/uid';

/**
 * @typedef {Object} ConnectorPoint
 * @property {number} x
 * @property {number} y
 * @property {string} _id
 */

/**
 * Generate connector
 *
 * @param {number} x
 * @param {number} y
 * @returns {ConnectorPoint}
 */
export const generateConnectorPoint = (x, y) => ({
  x,
  y,
  _id: uid(),
});

/**
 * Generate connector points by rectangle
 *
 * @param {Rect} rect
 * @param {number} count
 * @returns {ConnectorPoint[]}
 */
export const generateConnectorsByRect = (rect, count) => {
  const { width, height, rx, ry, x, y } = rect;
  const sideConnectorsCount = Math.round(count / 4);
  const sideSpacesCount = sideConnectorsCount + 1;
  const startX = x + rx;
  const startY = x + ry;

  const widthConnectorOffset = (width - rx * 2) / (sideSpacesCount);
  const heightConnectorOffset = (height - ry * 2) / (sideSpacesCount);

  const sideConnectors = range(1, sideConnectorsCount + 1);

  const topConnectors = sideConnectors.map(index => generateConnectorPoint(
    startX + widthConnectorOffset * index,
    y,
  ));

  const rightConnectors = sideConnectors.map(index => generateConnectorPoint(
    x + width,
    startY + heightConnectorOffset * index,
  ));

  const bottomConnectors = sideConnectors.map(index => generateConnectorPoint(
    startX + widthConnectorOffset * index,
    y + height,
  ));

  const leftConnectors = sideConnectors.map(index => generateConnectorPoint(
    x,
    startY + heightConnectorOffset * index,
  ));

  const connectors = [
    ...topConnectors,
    ...rightConnectors,
    ...bottomConnectors,
    ...leftConnectors,
  ];

  if (!rx && !ry) {
    /* Angles connectors */
    connectors.push(
      generateConnectorPoint(x, y),
      generateConnectorPoint(x + width, y),
      generateConnectorPoint(x + width, y + height),
      generateConnectorPoint(x, y + height),
    );
  }

  return connectors;
};
