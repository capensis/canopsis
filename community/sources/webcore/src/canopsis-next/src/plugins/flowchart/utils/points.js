/**
 * Calculate points with ghosts points
 *
 * @param {Array} points
 * @returns {Array}
 */
export const getPointsWithGhosts = points => points.reduce((acc, point, index) => {
  const nextIndex = index + 1;
  const nextPoint = points[nextIndex];

  acc.push({ ...point, index });

  if (nextPoint) {
    acc.push({
      x: (point.x + nextPoint.x) / 2,
      y: (point.y + nextPoint.y) / 2,
      ghost: true,
      index: nextIndex,
    });
  }

  return acc;
}, []);
