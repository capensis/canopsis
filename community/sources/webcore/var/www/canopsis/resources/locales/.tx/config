[main]
host = https://www.transifex.com

[canopsis.global]
source_lang = en
source_file = lang-en.po
file_filter = lang-<lang>.po
type = PO
