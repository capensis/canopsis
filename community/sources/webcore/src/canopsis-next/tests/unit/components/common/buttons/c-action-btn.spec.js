import Faker from 'faker';

import { PAGINATION_TOTAL_VISIBLE } from '@/config';

import { mount, createVueInstance } from '@/unit/utils/vue';

import CActionBtn from '@/components/common/buttons/c-action-btn.vue';

const localVue = createVueInstance();

const mocks = {
  $config: {
    PAGINATION_TOTAL_VISIBLE,
  },
};

const stubs = {
  'v-btn': {
    template: `
      <button
        class="v-btn"
        @click="$listeners.click"
      >
        <slot />
      </button>
    `,
  },
};

const factory = (options = {}) => mount(CActionBtn, {
  localVue,
  stubs,
  mocks,
  ...options,
});

describe('c-action-btn', () => {
  it('Action button with default type and custom icon.', () => {
    const icon = Faker.datatype.string();
    const wrapper = factory({ propsData: { icon, type: 'edit' } });

    const iconElement = wrapper.find('v-icon-stub');

    expect(iconElement.text()).toBe(icon);
  });

  it('Action button with default type and custom tooltip.', () => {
    const tooltip = Faker.datatype.string();
    const wrapper = factory({ propsData: { tooltip, type: 'duplicate' } });

    const tooltipContentElement = wrapper.find('span');

    expect(tooltipContentElement.text()).toBe(tooltip);
  });

  it('Action button with default type and custom color.', () => {
    const color = Faker.datatype.string();
    const wrapper = factory({ propsData: { color, type: 'duplicate' } });

    const iconElement = wrapper.find('v-icon-stub');

    expect(iconElement.attributes('color')).toBe(color);
  });

  it('Action button without default type and all custom props.', () => {
    const icon = Faker.datatype.string();
    const tooltip = Faker.datatype.string();
    const color = Faker.datatype.string();
    const wrapper = factory({ propsData: { color, icon, tooltip } });

    const iconElement = wrapper.find('v-icon-stub');
    const tooltipContentElement = wrapper.find('span');

    expect(iconElement.text()).toBe(icon);
    expect(tooltipContentElement.text()).toBe(tooltip);
    expect(iconElement.attributes('color')).toBe(color);
  });

  it('Check loading property.', () => {
    const wrapper = factory({ propsData: { loading: true } });

    const buttonElement = wrapper.find('button.v-btn');

    expect(buttonElement.attributes('loading')).toBeTruthy();
  });

  it('Check disabled property.', () => {
    const wrapper = factory({ propsData: { disabled: true } });

    const buttonElement = wrapper.find('button.v-btn');

    expect(buttonElement.attributes('disabled')).toBeTruthy();
  });

  it('Check button slot.', () => {
    const wrapper = factory({
      slots: {
        button: '<div class="name-slot" />',
      },
    });

    const slotElement = wrapper.find('div.name-slot');

    expect(slotElement.exists()).toBeTruthy();
  });

  it('Click event working correctly.', () => {
    const onClick = jest.fn();
    const wrapper = factory({
      listeners: {
        click: onClick,
      },
    });

    const button = wrapper.find('button.v-btn');

    button.trigger('click');

    expect(onClick).toHaveBeenCalledTimes(1);
  });

  it('Renders `c-action-btn` with default edit type correctly.', () => {
    const wrapper = mount(CActionBtn, {
      localVue,
      mocks,
      propsData: { type: 'edit' },
    });

    expect(wrapper.element).toMatchSnapshot();
  });

  it('Renders `c-action-btn` with default edit type correctly.', () => {
    const wrapper = mount(CActionBtn, {
      localVue,
      mocks,
      propsData: { type: 'duplicate' },
    });

    expect(wrapper.element).toMatchSnapshot();
  });

  it('Renders `c-action-btn` with default edit type correctly.', () => {
    const wrapper = mount(CActionBtn, {
      localVue,
      mocks,
      propsData: { type: 'delete' },
    });

    expect(wrapper.element).toMatchSnapshot();
  });

  it('Renders `c-action-btn` with custom type correctly.', () => {
    const wrapper = mount(CActionBtn, {
      localVue,
      mocks,
      propsData: { icon: 'test_icon', color: 'color', tooltip: 'tooltip' },
    });

    expect(wrapper.element).toMatchSnapshot();
  });
});
