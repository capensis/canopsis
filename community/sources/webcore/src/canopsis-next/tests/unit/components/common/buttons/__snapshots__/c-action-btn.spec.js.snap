// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`c-action-btn Renders \`c-action-btn\` with custom type correctly. 1`] = `
<v-tooltip-stub
  bottom="true"
  closedelay="200"
  contentclass=""
  debounce="0"
  fixed="true"
  maxwidth="auto"
  nudgebottom="0"
  nudgeleft="0"
  nudgeright="0"
  nudgetop="0"
  nudgewidth="0"
  opendelay="200"
  tag="span"
>
  <template>
    <v-btn-stub
      activeclass="v-btn--active"
      class="mx-1"
      icon="true"
      tag="button"
      type="button"
    >
      <v-icon-stub
        color="color"
      >
        test_icon
      </v-icon-stub>
    </v-btn-stub>
  </template>
  <span>
    tooltip
  </span>
</v-tooltip-stub>
`;

exports[`c-action-btn Renders \`c-action-btn\` with default edit type correctly. 1`] = `
<v-tooltip-stub
  bottom="true"
  closedelay="200"
  contentclass=""
  debounce="0"
  fixed="true"
  maxwidth="auto"
  nudgebottom="0"
  nudgeleft="0"
  nudgeright="0"
  nudgetop="0"
  nudgewidth="0"
  opendelay="200"
  tag="span"
>
  <template>
    <v-btn-stub
      activeclass="v-btn--active"
      class="mx-1"
      icon="true"
      tag="button"
      type="button"
    >
      <v-icon-stub>
        edit
      </v-icon-stub>
    </v-btn-stub>
  </template>
  <span>
    common.edit
  </span>
</v-tooltip-stub>
`;

exports[`c-action-btn Renders \`c-action-btn\` with default edit type correctly. 2`] = `
<v-tooltip-stub
  bottom="true"
  closedelay="200"
  contentclass=""
  debounce="0"
  fixed="true"
  maxwidth="auto"
  nudgebottom="0"
  nudgeleft="0"
  nudgeright="0"
  nudgetop="0"
  nudgewidth="0"
  opendelay="200"
  tag="span"
>
  <template>
    <v-btn-stub
      activeclass="v-btn--active"
      class="mx-1"
      icon="true"
      tag="button"
      type="button"
    >
      <v-icon-stub>
        file_copy
      </v-icon-stub>
    </v-btn-stub>
  </template>
  <span>
    common.duplicate
  </span>
</v-tooltip-stub>
`;

exports[`c-action-btn Renders \`c-action-btn\` with default edit type correctly. 3`] = `
<v-tooltip-stub
  bottom="true"
  closedelay="200"
  contentclass=""
  debounce="0"
  fixed="true"
  maxwidth="auto"
  nudgebottom="0"
  nudgeleft="0"
  nudgeright="0"
  nudgetop="0"
  nudgewidth="0"
  opendelay="200"
  tag="span"
>
  <template>
    <v-btn-stub
      activeclass="v-btn--active"
      class="mx-1"
      icon="true"
      tag="button"
      type="button"
    >
      <v-icon-stub
        color="error"
      >
        delete
      </v-icon-stub>
    </v-btn-stub>
  </template>
  <span>
    common.delete
  </span>
</v-tooltip-stub>
`;
